packages/go_router/lib/src/state.dart

```diff
@@ -24,13 +24,12 @@ class GoRouterState {
     this.extra,
     this.error,
     ValueKey<String>? pageKey,
-  })  : pageKey = pageKey ??
+  }) : pageKey = pageKey ??
             ValueKey<String>(error != null
                 ? 'error'
                 : fullpath != null && fullpath.isNotEmpty
                     ? fullpath
-                    : subloc),
-        assert((path ?? '').isEmpty == (fullpath ?? '').isEmpty);
```

Comment:
Maybe keep this assert for GoRoute?

assert(route is! GoRoute || ....)

Reply:
Can you elaborate? I'm not sure what you want to assert here.

Reply:
I don't know what I was talking about neither, this is probably fine.